backend/llm/openai.py

```diff
@@ -216,15 +216,9 @@ async def wrap_done(fn: Awaitable, event: asyncio.Event):
             finally:
                 event.set()
 
-        # Use the acall method to perform an async call to the QA chain
         task = asyncio.create_task(
             wrap_done(
-                qa.acall(
-                    {
-                        "question": question,
-                        "chat_history": transformed_history,
-                    }
-                ),
+                self.qa._acall_chain(self.qa, question, transformed_history),
                 callback.done,
```

Comment:
What is callback type ? Is it a class here ? It's usually a function

Reply:
Its a class. https://github.com/hwchase17/langchain/blob/master/langchain/callbacks/streaming_aiter.py

The type is hard because they are all very different. Specifically we use the AsyncIteratorCallbackHandler. It inherits from the AsyncCallbackHandler. https://github.com/hwchase17/langchain/blob/master/langchain/callbacks/base.py#L258